script/release-notes

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+# Usage: script/release-notes
+#
+# - fetches latest tags
+# - switches to master branch
+# - ensures there are unreleased features
+# - creates .git/RELNOTES file from git history
+# - opens .git/RELNOTES in $EDITOR (default: vim)
+# - exits with vim's exit status (and non-empty RELNOTES file)
+
+set -euo pipefail
+IFS=$'\n\t'
+
+abort() {
+  printf 'Aborting: %s\n' "$1" >&2
+  exit 1
+}
+
+assert_features_since() {
+  local previous_tag=$1
+
+  if git diff --quiet "${previous_tag}..HEAD" -- install.sh bin contrib libexec; then
+    abort "No features to release since '${previous_tag}'"
+  fi
+}
+
+changes_since() {
+  local previous_tag=$1
+  git log --first-parent --format='%w(0,0,2)* %B%n' --reverse "${previous_tag}..HEAD^" --  install.sh bin contrib libexec
+}
+
+
+git fetch --quiet --tags
+
+git checkout --quiet master
+
+# This only finds tags reachable from HEAD
+PREV_TAG="$(git describe --tags --match 'v*' --abbrev=0)"
+RELNOTES="$(git rev-parse --git-dir)/RELNOTES"
```

Comment:
Are you putting the `RELNOTES` in the `.git` directory to hide it from `git diff`? Perhaps it'd be better to `.gitignore` it instead.

Reply:
Not specifically to hide it, but rather because the RELNOTES file is temporary working file. I was modeling it after how git itself writes `COMMIT_MSG` (can't remember the exact filename) to `.git/`as well as how `hub` writes its `PR_MSG` there as well. Again, unsure on the exact filename for each of those, but both `git` and `hub` write their files to .git until the full operation is completed at which point, they are cleaned up.

Reply:
If you consider just those 2 use cases a convention. 🤷‍♂️ was enough of a pattern that I thought made a good excuse to keep it out of repo root 😸

Reply:
Well, as use cases go, `git` and `hub` seem fairly mainstream enough to follow. 😉